frontend/pages/compensators/[documentId].js

```diff
@@ -0,0 +1,498 @@
+                            />
+                          )
+                        : sdgData.find(
+                            (defaultSDG) => defaultSDG.title === sdg.name
+                          )?.icon || TreePine,
+                    }))}
+                  />
+                </div>
+              </div>
+            </div>
+            {(compensator?.compensationProgressDescription ||
+              compensator?.compensationProgressImage?.url) && (
+              <div>
+                <div>
+                  <h2 className="text-gray-800 font-bold mb-4 border-t border-blue-100 mt-6 pt-6">
+                    SOFTA WATER FOOTPRINT COMPENSATION & REDUCTION 5 YEAR
+                    PROGRESS:
+                  </h2>
+
+                  {imageUrl && (
+                    <div className="relative w-full max-w-lg mx-auto mb-4">
+                      <Image
+                        src={imageUrl}
+                        alt={compensator.name}
+                        className="object-contain rounded-lg shadow-md"
+                        unoptimized
+                        width={1200}
+                        height={300}
+                      />
+                    </div>
+                  )}
+                  <p className="pt-2">
+                    {compensator?.compensationProgressDescription}
+                  </p>
+                </div>
+              </div>
+            )}
+          </div>
+
+          {projects.length > 0 && (
+            <section className="py-8">
+              <div className="max-w-6xl mx-auto">
+                <div className="flex justify-between mb-6 items-center">
+                  <h2 className="text-4xl font-bold text-[#0DA2D7]">
+                    Projects Supported
+                  </h2>
+
+                  <div className="flex gap-2">
+                    {isMapView ? (
+                      <button
+                        className="p-1 text-gray-500 hover:text-gray-700 cursor-pointer"
+                        onClick={() => setIsMapView(!isMapView)}
+                      >
+                        <List className="w-6 h-6" />
+                      </button>
+                    ) : (
+                      <button
+                        className="p-1 text-gray-500 hover:text-gray-700 cursor-pointer"
+                        onClick={() => setIsMapView(!isMapView)}
+                      >
+                        <MapPinned className="w-6 h-6" />
+                      </button>
+                    )}
+                  </div>
+                </div>
+                {!isMapView ? (
+                  <div>
+                    <ProjectsListing projects={projects} />
+                  </div>
+                ) : (
+                  <Map projectIds={projectsId} />
+                )}
+              </div>
+            </section>
+          )}
+        </div>
+      </div>
+    );
+  };
+
+  return renderContent();
+}
+
+const ProjectsListing = ({ projects }) => {
```

Comment:
This should be it's own component. In general I pretty strongly believe in one component per file.